src/java/org/apache/cassandra/cql3/statements/schema/AlterTableStatement.java

```diff
@@ -703,6 +705,79 @@ private void validateCanDropCompactStorage()
         }
     }
 
+    public static class DropConstraint extends AlterTableStatement
+    {
+        final ColumnIdentifier constraintName;
+
+        DropConstraint(String keyspaceName, String tableName, boolean ifTableExists, ColumnIdentifier constraintName)
+        {
+            super(keyspaceName, tableName, ifTableExists);
+            this.constraintName = constraintName;
+        }
+
+        @Override
+        public KeyspaceMetadata apply(Epoch epoch, KeyspaceMetadata keyspace, TableMetadata table, ClusterMetadata metadata)
+        {
+            boolean removed = false;
+            TableMetadata.Builder tableBuilder = table.unbuild().epoch(epoch);
+            for(CqlConstraint constraint : tableBuilder.constraints())
+            {
+                if (constraint.constraintName == constraintName)
+                {
+                    tableBuilder.removeConstraint(constraint);
+                    removed = true;
```

Comment:
don't you need to `break`? If you found such constrain, you are done, you do not need to iterate to the end.